### DISCUSSION TITLE
How to suppress compiler warnings?

### DISCUSSION THREAD

**Discussion by ntreeby** [vscode-abl/vscode-abl#28] [Q&A] [answered]
Is there an option to suppress compiler warnings?  In particular, my project is seeing a lot of "line will not be reached" warnings coming out of ADM2 library code I extracted to %DLC%\src\adm2.

**Comment by gquerret** [comment, marked as answer]
You may try adding `-swl ...` to the `extraParameters` attribute of the JSON configuration file.

**Reply by ntreeby** [reply]
That worked, though it must be added that the Progress docs (11.7 & 12) give the wrong error#  for "statement will not be reached" - it is 15090, not 1509 as the docs state.  Also the file must be on the PROPATH, or give an absolute path, unlike say a .ini or .pf which can be relative to the working directory.  There is also -sw which overrides -swl and suppresses all warnings, no need for a file.

A useful feature implemented in an odd way.
